### DISCUSSION TITLE
Step by Step installation guide

### DISCUSSION THREAD

**Discussion by 2003flo** [bnhf/openvpn-admin-plus#51] [Q&A] [answered]
Hello!
I have already written to you on reddit. I have a working pivpn installation, but I can't get this container to work.

What I have done so far:
- I have a working pivpn installation.
- I added the following line to server.conf: management localhost 2080
- I created a stack, with the details from your screenshot: https://github.com/bnhf/openvpn-admin-plus#Prod . Here I adjusted the variables:

```
OPENVPN_ADMIN_USERNAME=admin
 OPENVPN_ADMIN_PASSWORD=b3secure
 COUNTRY= I don't know what is used in EasyRSA. I have specified something arbitrary here.
 PROVINCE= see Country
 CITY= see Country
 ORG= see Country
 EMAIL= Fake@mail.com
 OU= fake
 PIVPN_SERVER=xyz (name from server.conf)
 PIVPN_CONF=server.conf
 TZ=see Country
```

The container is then successfully created and I can access the web interface. Under Logs I can also see the connection to my server. But on the status page it says "Fix your configuration" and the OpenVPN version is not displayed.

I hope you can help.

**Comment by bnhf** [comment, marked as answer]
@2003flo 

> * I added the following line to server.conf: management localhost 2080

You can't use localhost 2080 here.

You have two options: the first is to use 0.0.0.0 2080 in server.conf, and the IP address of your OpenVPN server followed by :2080 in Configuration-Settings-Management Interface Address. (for example: 192.168.1.100:2080)

The other option, which is better as it doesn't expose the management interface to your LAN, is to use 172.17.0.1 2080 in server.conf and 172.17.0.1:2080 in Settings. This is the IP of the default Docker Gateway, so all interaction with the Management Interface happens through this limited gateway.

These values are suggested when you look at the help blocks under each data item in Configuration - OpenVPN Settings and Configuration - Settings.

Be sure to make the change to your server.conf file first (best done by duplicating the initial server.conf directives in the WebUI) and reboot or restart the OpenVPN service. Once you've done this once the values will be stored in the database.

Also, just to confirm, you are using OpenVPN 2.5.x or higher right?

**Comment by bnhf** [comment]
@2003flo 

Here's what I'm talking about from Configuration - Settings:

![screenshot-nuc10-pc2-2023 02 06-11_05_03](https://user-images.githubusercontent.com/41088895/217050543-58832e4b-476a-4dc7-824b-9515b674a120.png)

**Comment by bnhf** [comment]
@2003flo 

And from Configuration - OpenVPN Config:

![screenshot-nuc10-pc2-2023 02 06-11_07_18](https://user-images.githubusercontent.com/41088895/217050866-df2c9b96-0ebe-4a09-8f06-179a6fa8d16c.png)

Be aware all of the values need to be correct the first time you create a server.conf using the WebUI. Have the PiVPN created server.conf file open in another window so you can copy and paste as needed. After that first time through, your OpenVPN-Admin-Plus database values will match the actual server.conf file.

**Comment by 2003flo** [comment]
Thank you, now it works.

I have quite a few other questions. Should I ask them in this thread or open new ones?

**Comment by bnhf** [comment]
@2003flo 

Here is fine.

**Comment by 2003flo** [comment]
1. Status page: Is there an automatic reload function for the status page?

2. Status page: Is it possible to change the size of KB Received / KB Sent? E.g. in Megabyte?

3. Settings page: What is the field "Server Address External" for?

4. OpenVPN Config page: Do I see it right that I don't have to change anything here if I just want to continue my existing configuration? 

5. OpenVPN Config Page: There is no free text field for the complete server.conf, is there?

6. Certificates: On the status page I see that client xy is connected. Why does this not show up in the certificates as well?

**Reply by bnhf** [reply]
@2003flo 

> 1. Status page: Is there an automatic reload function for the status page?

Status page refresh is fixed at 60 seconds

> 2. Status page: Is it possible to change the size of KB Received / KB Sent? E.g. in Megabyte?

Overall client data in/out is in MB and individual client data in KB. We tried a few combinations before settling on this one.

> 3. Settings page: What is the field "Server Address External" for?

That's the address that will be used when you generate client .ovpn files. Your static public IP, or DDNS address if your IP is Dynamic.

> 4. OpenVPN Config page: Do I see it right that I don't have to change anything here if I just want to continue my existing configuration?

Part of the initial setup is to enter your server.conf data once. Don't skip this or you'll have a variety of issues.

> 5. OpenVPN Config Page: There is no free text field for the complete server.conf, is there?

No. But I think you'll find that many of the fields already have the correct data. Being able to change the server.conf file through the WebUI is part of what OpenVPN Admin Plus is all about.

> 6. Certificates: On the status page I see that client xy is connected. Why does this not show up in the certificates as well?

Not sure I totally understand this question. The status page shows connected clients. The certificates page is for managing client credentials -- i.e. creating .ovpn or .conf files for your clients to use to connect. Certificates can be revoked and then removed if you want to discontinue someone's access. Does that answer it?

**Reply by 2003flo** [reply]
Thanks for your help. I‘ll post my config later. First I Continue with my questions:

> > 4. OpenVPN Config page: Do I see it right that I don't have to change anything here if I just want to continue my existing configuration?
> 
> Part of the initial setup is to enter your server.conf data once. Don't skip this or you'll have a variety of issues.

So I should define this parameter also if I neither want to change anything in the config nor create new users??


> > 6. Certificates: On the status page I see that client xy is connected. Why does this not show up in the certificates as well?
> 
> Not sure I totally understand this question. The status page shows connected clients. The certificates page is for managing client credentials -- i.e. creating .ovpn or .conf files for your clients to use to connect. Certificates can be revoked and then removed if you want to discontinue someone's access. Does that answer it?

No, that does not answer my question. I was wondering why a user who is connected according to the Status page is not listed on the Certificates page.

One more question:
7. What is the meaning of the parameters Country, Province, etc.? What do I specify here?

**Reply by bnhf** [reply]
@2003flo 

> So I should define this parameter also if I neither want to change anything in the config nor create new users??

If you don't do the initial configuration for OpenVPN config, then that would be the result -- no accurate certificate creation for new users, no ability to change single elements of the server.conf, and I don't believe you'd get persistent log data.

This would be an unusual way to use OpenVPN Admin Plus, and I wouldn't recommend it -- though the status page should still be accurate (untested for use in this way though).

> No, that does not answer my question. I was wondering why a user who is connected according to the Status page is not listed on the Certificates page.

Any users created through this WebUI or PiVPN should show up in the list of valid certificates. If they don't there's something amiss with your installation. How was this user created?

> What is the meaning of the parameters Country, Province, etc.? What do I specify here?

These parameters slightly improve the EasyRSA cryptography, through additional "randomization".  They're detailed here in the README:

![screenshot-github com-2023 02 07-08_00_30](https://user-images.githubusercontent.com/41088895/217281152-9b0d7d25-3015-40b6-8051-94b01db77abc.png)

**Reply by 2003flo** [reply]
The user was created with pivpn.

In fact, I want to use the tool only for monitoring. I want to know who is online, what bandwidth is/was used. And preferably a history.

**Comment by bnhf** [comment]
@2003flo 

Making a backup of your original server.conf file is always a good idea, especially if you're not super familiar with OpenVPN's directives. Don't name it anything with .conf though -- I usually use server.pivpn for my backup. If you're new to Linux, having Cockpit installed with the Navigator plugin is really handy for "system" tasks. I used the command line for so many decades that one of my missions in life is to use it as little as possible from here forward. ;-)

**Comment by bnhf** [comment]
@2003flo 

If it'd help, send me a screenshot of your completed Server - Config (blur the port if you know how -- but even that's not really critical), and I'll tell you if I see any potential issues. Be sure to expand the Extra Server and Client Options so I can see everything in those fields.

**Comment by karabelnikov** [comment]
@2003flo
> Certificates: On the status page I see that client xy is connected. Why does this not show up in the certificates as well?

The Certificates page is intended for issuing, revoking and deleting certificates from the database, not for monitoring! This page also allows you to download ready-made configurations (.zip or .ovpn) for your OpenVPN Client.

**Important!** Using the certificate removal function from the database, keep in mind that it is deleted from the file index.txt which is necessary for forming a revoke list. If you delete a certificate from the database, it does not get into the new revocation list.

**Comment by 2003flo** [comment]
I am using openvpn-admin-plus now a few days. I am happy with the fact that I can watch the current users and their traffic on the status page.

However, I am surprised that even after restarting the whole computer, my log continues to show data from 06 February 2023, although today is 14 February.

**Reply by bnhf** [reply]
The log will get setup correctly once you run through entering data into Configuration - OpenVPN Config the first time.

The OpenVPN Admin Plus database will be populated, and a server.conf file with your directives, plus a few needed for the full WebUI functionality will be generated.

Have your current server.conf open in another window as you're entering the data. Many fields will already be correct. Save & Apply followed by an OpenVPN restart or system reboot and you'll be good to go.

Saving a copy of your current server.conf as server.pivpn first is always a good idea.

**Reply by 2003flo** [reply]
I hope I have understood you correctly?

I have not used openvpn config and do not want to use it because I do not want to change my server.conf. This has the consequence that log does not work?

**Reply by bnhf** [reply]
You would have had to modify server.conf to even get this far, by adding a "management" directive.

This is similar, and will be done for you through the WebUI. Do you prefer to modify server.conf manually?

**Reply by bnhf** [reply]
There are a few log file directives that need to be added or modified to give Docker access to the log.

This can also be done by directly editing server.conf, as you did to get the management directive in there.

Doing it via the WebUI is simply intended to make it easier, and would have taken care of all this for you.

**Reply by 2003flo** [reply]
Yes, I already modified the server.conf manually. Here is my server.conf:

```
management 172.17.0.1 2080
dev tun
proto udp
port secret
ca /etc/openvpn/easy-rsa/pki/ca.crt
cert /etc/openvpn/easy-rsa/pki/issued/secret.crt
key /etc/openvpn/easy-rsa/pki/private/secret.key
dh none
push "redirect-gateway def1"
ecdh-curve prime256v1
topology subnet
server 10.8.0.0 255.255.255.0
# Set your primary domain name server address for clients
push "dhcp-option DNS secret"
push "block-outside-dns"
# Override the Client default gateway by using 0.0.0.0/1 and
# 128.0.0.0/1 rather than 0.0.0.0/0. This has the benefit of
# overriding but not wiping out the original default gateway.
client-to-client
client-config-dir /etc/openvpn/ccd
keepalive 15 120
remote-cert-tls client
tls-version-min 1.2
tls-crypt /etc/openvpn/easy-rsa/pki/ta.key
cipher AES-256-CBC
auth SHA256
user openvpn
group openvpn
persist-key
persist-tun
crl-verify /etc/openvpn/crl.pem
status /var/log/openvpn-status.log 20
status-version 3
syslog
verb 3
#DuplicateCNs allow access control on a less-granular, per user basis.
#Remove # if you will manage access by user instead of device.
#duplicate-cn
# Generated for use by PiVPN.io
#
```


And here are the configuration pages of openvpn-admin-plus:

![image](https://user-images.githubusercontent.com/63111038/219068178-05cc8a44-bd55-4e2f-ac5c-dc1946dc13ee.png)

![image](https://user-images.githubusercontent.com/63111038/219068713-3ea0c443-dd40-47f9-a214-c7545d755ef9.png)

![image](https://user-images.githubusercontent.com/63111038/219068996-17e8675d-b351-4ec3-beb3-619caaaa605e.png)

![image](https://user-images.githubusercontent.com/63111038/219069415-eab2162b-cd22-4ea7-b86e-da8980a9946e.png)

**Comment by bnhf** [comment]
@2003flo 

The section in your server.conf file that would be updated is:

```console
# status /var/log/openvpn-status.log 20
# status-version 3
# syslog
log /etc/openvpn/openvpn.log
verb 3
mute 10
```

This can, of course, be done manually too. :-) Restart OpenVPN or reboot after updating.

**Reply by 2003flo** [reply]
Thank you very much. Now the log works again.

I didnt comment out the status command, because I'm playing around with grafana.

Now that you see my settings, do you have any idea why a connected user is not listed under Certificates?

**Reply by bnhf** [reply]
Nothing is jumping out at me. Could you post the output of "sudo tree" (you'll probably need to install tree first) from the /etc/openvpn directory, and the contents of /etc/openvpn/easy-rsa/pki/index.txt?
